notebooks/notebook_template_review.py

```diff
@@ -193,71 +194,81 @@ def parse_notebook(path: str) -> None:
+        LinksRule().validate()
         
-        OverviewRule().validate(path, cells)
+        OverviewRule().validate()
        
-        ObjectiveRule().validate(path, cells)
-        if NotebookRule.desc != '':
-            add_index(path, 
-                      tag, 
-                      NotebookRule.title, 
-                      NotebookRule.desc, 
-                      NotebookRule.uses, 
-                      NotebookRule.steps, 
-                      NotebookRule.git_link, 
-                      NotebookRule.colab_link, 
-                      NotebookRule.workbench_link
-            )
+        ObjectiveRule().validate()
             
-        RecommendationsRule().validate(path, cells)
+        RecommendationsRule().validate()
 
-        DatasetRule().validate(path, cells)
+        DatasetRule().validate()
 
-        CostsRule().validate(path, cells)
+        CostsRule().validate()
 
-        SetupLocalRule().validate(path, cells)
+        SetupLocalRule().validate()
 
-        HelpersRule().validate(path, cells)
+        HelpersRule().validate()
 
-        InstallationRule().validate(path, cells)
+        InstallationRule().validate()
 
-        RestartRule().validate(path, cells)
+        RestartRule().validate()
         
-        VersionsRule().validate(path, cells)
+        VersionsRule().validate()
         
-        BeforeBeginRule().validate(path, cells)
+        BeforeBeginRule().validate()
         
-        EnableAPIsRule().validate(path, cells)
+        EnableAPIsRule().validate()
         
-        SetupProjectRule().validate(path, cells)
+        SetupProjectRule().validate()
+        if NotebookRule.desc != '':
+            add_index(path, 
+                      tag, 
+                      NotebookRule.title, 
+                      NotebookRule.desc, 
+                      NotebookRule.uses, 
+                      NotebookRule.steps, 
+                      NotebookRule.git_link, 
+                      NotebookRule.colab_link, 
+                      NotebookRule.workbench_link
+            )
 
 
 class NotebookRule(ABC):
     """
     Abstract class for defining notebook conformance rules
     """
-    cell_index = 0
-    title = ''
-    git_link = None
-    colab_link = None
-    workbench_link = None
-    desc = ''
-    uses = ''
-    steps = ''
-    costs = []
+    path = None     # The relative path to the notebook
+    cell_index = 0  # The index of the last cell that was parsed (reviewed).
+    title = ''      # The H1 title for the notebook
+    git_link = None  # The GitHub link
+    colab_link = None  # The Colab link
+    workbench_link = None  # The Workbench link
+    desc = ''  # The description in the objective section
+    uses = ''  # The resources used in the objective section
+    steps = ''  # The steps in the objective section
+    costs = []  # The paid services used as derived from the resources used subsection
     
     @staticmethod
-    def init():
+    def init(path: str,
```

Comment:
You shouldn't need to define an `init` method in your abstractclass. The subclasses will define their own.

Just describe the methods in your ABC.

Reply:
done